src/auth/views/auth/approve-login/approve-login.ts

```diff
@@ -0,0 +1,66 @@
+import { Component } from 'vue-property-decorator';
+import { authOnLogin, redirectToDashboard } from '../../../../_common/auth/auth.service';
+import AppLoading from '../../../../_common/loading/loading.vue';
+import { AppProgressPoller } from '../../../../_common/progress/poller/poller';
+import { BaseRouteComponent } from '../../../../_common/route/route-component';
+
+@Component({
+	name: 'RouteApproveLogin',
+	components: {
+		AppProgressPoller,
+		AppLoading,
+	},
+})
+export default class RouteApproveLogin extends BaseRouteComponent {
+	pollingToken: string | null = null;
+
+	isExpired = false;
+	isRejected = false;
+
+	get routeTitle() {
+		return this.$gettext('Approve Login Location');
+	}
+
+	get isPolling() {
+		return !this.isExpired && !this.isRejected;
+	}
+
+	routeCreated() {
+		if (!GJ_IS_SSR) {
+			this.pollingToken = sessionStorage.getItem('login-polling-token');
+			if (!this.pollingToken) {
+				this.$router.push({ name: 'auth.login' });
+				return;
+			}
+
+			sessionStorage.removeItem('login-polling-token');
+		}
+	}
+
+	async onApproved() {
+		// TODO: the approved login flow does not respect AppAuthLogin's redirectTo property
```

Comment:
Was there still work that needed to be done with this, or is it fine to just leave it as a comment?